astrokat/observe_main.py

```diff
@@ -220,14 +220,34 @@ def cadence_target(target_list):
 
 def above_horizon(katpt_target, horizon=20.0, duration=0.0):
     """Check target visibility."""
-    [azim, elev] = katpt_target.azel(time.time())
+    # use local copies so you do not overwrite target time attribute
+    if 'azel' in katpt_target.tags:
+        return True
+
+    # must be celestial target (ra, dec)
+    local_target = katpt_target.body.copy()
+    local_observer = katpt_target.antenna.observer.copy()
+
+    # check that target is visible at start of track
+    start_ = timestamp2datetime(time.time())
+    local_observer.date = ephem.date(start_)
+    local_target.compute(local_observer)
+    [azim, elev] = [local_target.az, local_target.alt]
+    user_logger.trace(
+        "TRACE: target at start (az, el)= ({}, {})".format(azim, elev)
+    )
     if not elev > ephem.degrees(str(horizon)):
         return False
 
+    # check that target will be visible at end of track
     if duration:
-        [azim, elev] = katpt_target.azel(time.time() + duration)
-        # dummy call to reset timestamp for logger since azel() changes timestampe
-        [x, y] = katpt_target.azel(time.time())
+        end_ = timestamp2datetime(time.time() + duration)
+        local_observer.date = ephem.date(end_)
+        local_target.compute(local_observer)
+        [azim, elev] = [local_target.az, local_target.alt]
```

Comment:
This is almost identical to lines 232 to 235.  Suggest the common code is moved to a function, so we don't have to repeat ourselves.  
(Or use the katpoint.azel method)

Reply:
Cleaned up duplicate code, preference to using a utility function that explicitly requires ephem targets